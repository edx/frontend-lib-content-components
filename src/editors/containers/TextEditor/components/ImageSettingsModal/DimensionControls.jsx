import React from 'react';
import PropTypes from 'prop-types';
import {
  Form,
  Icon,
  IconButton,
} from '@edx/paragon';
import {
  Locked,
  Unlocked,
} from '@edx/paragon/icons';

import hooks from './hooks';

/**
 * Wrapper for image dimension inputs and the lock checkbox.
 * @param {obj} locked - locked dimension object
 * @param {func} lock - lock dimensions
 * @param {func} setHeight - updates dimensions based on new height
 * @param {func} setWidth - updates dimensions based on new width
 * @param {func} unlock - unlock dimensions
 * @param {func} updateDimensions - update dimensions callback
 * @param {obj} value - local dimension values { height, width }
 */
export const DimensionControls = ({
  locked,
  lock,
  setHeight,
  setWidth,
  unlock,
  updateDimensions,
  value,
}) => ((value !== null) && (
  <Form.Group>
    <Form.Label as="h4">Image Dimensions</Form.Label>
    <div className="mt-4.5">
      <Form.Control
        className="dimension-input"
        type="number"
        value={value.width}
        min={1}
        onChange={hooks.onInputChange(setWidth)}
        onBlur={updateDimensions}
        floatingLabel="Width"
      />
      <Form.Control
        className="dimension-input"
        type="number"
        value={value.height}
        min={1}
        onChange={hooks.onInputChange(setHeight)}
        onBlur={updateDimensions}
        floatingLabel="Height"
      />
      <IconButton
        className="d-inline-block"
        alt={locked ? 'unlock dimensions' : 'lock dimensions'}
        iconAs={Icon}
        src={locked ? Locked : Unlocked}
        onClick={locked ? unlock : lock}
      />
    </div>
  </Form.Group>
));
DimensionControls.defaultProps = {
  locked: null,
  value: {
    height: 100,
    width: 100,
  },
};
DimensionControls.propTypes = ({
  value: PropTypes.shape({
    height: PropTypes.number,
    width: PropTypes.number,
  }),
  setHeight: PropTypes.func.isRequired,
  setWidth: PropTypes.func.isRequired,
  locked: PropTypes.shape({
    width: PropTypes.number,
    height: PropTypes.number,
  }),
  lock: PropTypes.func.isRequired,
  unlock: PropTypes.func.isRequired,
  updateDimensions: PropTypes.func.isRequired,
});

export default DimensionControls;
