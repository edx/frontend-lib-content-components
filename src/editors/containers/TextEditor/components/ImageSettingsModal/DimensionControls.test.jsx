import React from 'react';
import { shallow } from 'enzyme';
import DimensionControls from './DimensionControls';

jest.mock('./hooks', () => ({
  onInputChange: (handler) => ({ 'hooks.onInputChange': handler }),
}));

describe('DimensionControls', () => {
  const props = {
    locked: { 'props.locked': 'lockedValue' },
    value: { width: 20, height: 40 },
  };
  beforeEach(() => {
    props.setWidth = jest.fn().mockName('props.setWidth');
    props.setHeight = jest.fn().mockName('props.setHeight');
    props.lock = jest.fn().mockName('props.lock');
    props.unlock = jest.fn().mockName('props.unlock');
    props.updateDimensions = jest.fn().mockName('props.updateDimensions');
  });
  describe('render', () => {
    test('snapshot', () => {
      expect(shallow(<DimensionControls {...props} />)).toMatchSnapshot();
    });
    test('null value: empty snapshot', () => {
      const el = shallow(<DimensionControls {...props} value={null} />);
      expect(el).toMatchSnapshot();
      expect(el.isEmptyRender()).toEqual(true);
    });
  });
});
