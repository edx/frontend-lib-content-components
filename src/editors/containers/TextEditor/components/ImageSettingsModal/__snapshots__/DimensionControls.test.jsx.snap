// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DimensionControls render null value: empty snapshot 1`] = `""`;

exports[`DimensionControls render snapshot 1`] = `
<Form.Group>
  <Form.Label
    as="h4"
  >
    Image Dimensions
  </Form.Label>
  <div
    className="mt-4.5"
  >
    <Form.Control
      className="dimension-input"
      floatingLabel="Width"
      min={1}
      onBlur={[MockFunction props.updateDimensions]}
      onChange={
        Object {
          "hooks.onInputChange": [MockFunction props.setWidth],
        }
      }
      type="number"
      value={20}
    />
    <Form.Control
      className="dimension-input"
      floatingLabel="Height"
      min={1}
      onBlur={[MockFunction props.updateDimensions]}
      onChange={
        Object {
          "hooks.onInputChange": [MockFunction props.setHeight],
        }
      }
      type="number"
      value={40}
    />
    <IconButton
      alt="unlock dimensions"
      className="d-inline-block"
      iconAs="Icon"
      onClick={[MockFunction props.unlock]}
      src={[MockFunction icons.Locked]}
    />
  </div>
</Form.Group>
`;

exports[`DimensionControls render unlocked dimensions 1`] = `
<Form.Group>
  <Form.Label
    as="h4"
  >
    Image Dimensions
  </Form.Label>
  <div
    className="mt-4.5"
  >
    <Form.Control
      className="dimension-input"
      floatingLabel="Width"
      min={1}
      onBlur={[MockFunction props.updateDimensions]}
      onChange={
        Object {
          "hooks.onInputChange": [MockFunction props.setWidth],
        }
      }
      type="number"
      value={20}
    />
    <Form.Control
      className="dimension-input"
      floatingLabel="Height"
      min={1}
      onBlur={[MockFunction props.updateDimensions]}
      onChange={
        Object {
          "hooks.onInputChange": [MockFunction props.setHeight],
        }
      }
      type="number"
      value={40}
    />
    <IconButton
      alt="lock dimensions"
      className="d-inline-block"
      iconAs="Icon"
      onClick={[MockFunction props.lock]}
      src={[MockFunction icons.Unlocked]}
    />
  </div>
</Form.Group>
`;
